hardly/handlers/distgit.py

```diff
@@ -61,26 +118,40 @@ def run(self) -> TaskResults:
                 "Not creating a dist-git MR from "
                 f"{self.target_repo}:{self.target_repo_branch}"
             )
-            return TaskResults(success=True, details={})
+            return TaskResults(success=True)
 
         if not self.package_config:
             logger.debug("No package config found.")
-            return TaskResults(success=True, details={})
+            return TaskResults(success=True)
 
-        logger.debug(f"About to create a dist-git MR from source-git MR {self.mr_url}")
+        if (
+            self.target_repo_branch
+            not in self.packit.dg.local_project.git_project.get_branches()
+        ):
+            msg = f"Downstream {self.target_repo}:{self.target_repo_branch} branch does not exist."
+            self.project.get_pr(int(self.mr_identifier)).comment(msg)
+            logger.info(msg)
+            return TaskResults(success=True)
 
-        source_project = self.service_config.get_project(url=self.source_project_url)
-        self.local_project = LocalProject(
-            git_project=source_project,
-            ref=self.data.commit_sha,
-            working_dir=self.service_config.command_handler_work_dir,
-        )
+        if self.dist_git_pr_model:
+            logger.info(
+                f"{self.source_git_pr_model} already has corresponding {self.dist_git_pr_model}"
+            )
+            if self.dist_git_pr:
+                if self.action == GitlabEventAction.closed.value:
+                    msg = f"[Source-git MR]({self.mr_url}) has been closed."
+                    self.dist_git_pr.close()
+                elif self.action == GitlabEventAction.reopen.value:
+                    msg = f"[Source-git MR]({self.mr_url}) has been reopened."
+                    # https://github.com/packit/ogr/pull/714
+                    # self.dist_git_pr.reopen()
+                else:
+                    logger.error(f"Unknown action {self.action}")
+                    return TaskResults(success=False)
+                logger.info(msg)
+                self.dist_git_pr.comment(msg)
```

Comment:
I hope folks won't find this as spamming but it's totally significant to inform about such an action in the downstream PR.

Reply:
Yes, and in most cases, you close the source-git MR only once, so it's just one comment.